### PR TITLE
.format() coverted to f-strings in registration & restoration subpackage

### PR DESCRIPTION
**Pull request by Larkinnjm1**
## Description

".format()" call on strings objects modified to f-strings within registration and restoration packages of library to partially address issue #5474.

files modified:
	modified:   skimage/registration/_masked_phase_cross_correlation.py
	modified:   skimage/registration/tests/test_masked_phase_cross_correlation.py
	modified:   skimage/restoration/_denoise.py

## Checklist

- Docstrings for all functions
- Benchmark in `./benchmarks`, if your changes aren't covered by an
  existing benchmark
- Unit tests
- Clean style in [the spirit of PEP8](https://www.python.org/dev/peps/pep-0008/)

## For reviewers

- Check that the PR title is short, concise, and will make sense 1 year
  later.
- Check that new functions are imported in corresponding `__init__.py`.
- Check that new features, API changes, and deprecations are mentioned in
      `doc/release/release_dev.rst`.

### CODE REVIEW

File: skimage/registration/tests/test_masked_phase_cross_correlation.py
Comment (mkcor) [inline skimage/registration/tests/test_masked_phase_cross_correlation.py:19]:
```suggestion
def test_deprecated_masked_register_translation():
```
Interesting... but is this related to the PR? I would expect this change to live (at least) in a separate commit!

File: skimage/registration/_masked_phase_cross_correlation.py
Comment (mkcor) [inline skimage/registration/_masked_phase_cross_correlation.py:174]:
```suggestion
                f'equal, but dimensions along axis {axis} are not.')
```
:wink:

File: skimage/restoration/_denoise.py
Comment (Larkinnjm1) [reply skimage/restoration/_denoise.py:637]: Thanks for the feedback this is what was in the original code base they held these strings in a tuple so I used the same object as a result. Looking at the `warn` object from `_warnings.py` in the `_ shared` directory there is no impact in using a string directly. However with the fstrings being in a tuple you don't have to include a back slash at the end of each new line of text to avoid getting an IndentationError. Hence putting it in a tuple is potentially less error prone. Is this the potential reason why the code was written like this originally does any of the OGs know?

File: skimage/restoration/_denoise.py
Comment (Larkinnjm1) [reply skimage/restoration/_denoise.py:662]: Per the same response above to the braketing (tuple) object issue.

File: skimage/registration/tests/test_masked_phase_cross_correlation.py
Comment (Larkinnjm1) [reply skimage/registration/tests/test_masked_phase_cross_correlation.py:19]: Good catch! No idea how that got deleted!  😅

File: skimage/restoration/_denoise.py
Comment (grlee77) [reply skimage/restoration/_denoise.py:635]: This is actually already just a string and not a tuple, since there is no comma within the parenthesis. I don't think there is a good reason to keep it this way and would just remove the extra parentheses.

Comment (Larkinnjm1) [reply skimage/restoration/_denoise.py:635]: Cool thanks for the input @grlee77 and @mkcor I will amend this item then.

**Review by grlee77 — APPROVED** [review]

File: skimage/restoration/_denoise.py
Comment (hmaarrfk) [inline skimage/restoration/_denoise.py:192]:
```suggestion
                raise ValueError('Bilateral filter is only implemented for '
                                 '2D grayscale images (image.ndim == 2) and '
                                 '2D multichannel (image.ndim == 3) images, '
```

File: skimage/restoration/_denoise.py
Comment (hmaarrfk) [inline skimage/restoration/_denoise.py:203]:
```suggestion
                msg = 'Input image must be grayscale, RGB, or RGBA; ' \
```

File: skimage/restoration/_denoise.py
Comment (hmaarrfk) [inline skimage/restoration/_denoise.py:197]:
```suggestion
                msg = 'The last axis of the input image is ' \
                      'interpreted as channels. Input image with '\
```

File: skimage/restoration/_denoise.py
Comment (hmaarrfk) [inline skimage/restoration/_denoise.py:200]:
```suggestion
                      'in last axis. ``denoise_bilateral``is implemented ' \
                      'for 2D grayscale and color images only.'
```

File: skimage/restoration/_denoise.py
Comment (hmaarrfk) [inline skimage/restoration/_denoise.py:635]:
```suggestion
        warn('Wavelet thresholding was designed for '
             'use with orthogonal wavelets. For nonorthogonal '
```

File: skimage/restoration/_denoise.py
Comment (hmaarrfk) [inline skimage/restoration/_denoise.py:637]:
```suggestion
             'likely to be suboptimal.')
```

File: skimage/restoration/_denoise.py
Comment (hmaarrfk) [inline skimage/restoration/_denoise.py:662]:
```suggestion
             'user-specified threshold will be ignored.')
```

File: skimage/restoration/_denoise.py
Comment (hmaarrfk) [inline skimage/restoration/_denoise.py:873]:
```suggestion
                         'methods are "BayesShrink" and "VisuShrink".')
```

File: skimage/restoration/_denoise.py
Comment (hmaarrfk) [inline skimage/restoration/_denoise.py:997]:
```suggestion
              'but channel_axis is None. If this is a color image, '\
```

File: skimage/restoration/_denoise.py
Comment (hmaarrfk) [inline skimage/restoration/_denoise.py:998]:
```suggestion
              'please set channel_axis=-1 for proper noise estimation.'
```

I think i prefer parenthesis to blackslashes. but that is rather minor.

**Review by hmaarrfk — COMMENTED** [review]
For concatenating strings, you only need an f on the portions that have fields to format.

**Comment by hmaarrfk** [comment]
Sorry for then noise. grlee already asked for a specific style. Lets follow it. This looks good to me.

**Review by hmaarrfk — APPROVED** [review]

File: skimage/registration/tests/test_masked_phase_cross_correlation.py
Comment (grlee77) [inline skimage/registration/tests/test_masked_phase_cross_correlation.py:11]:
```suggestion


```

File: skimage/registration/tests/test_masked_phase_cross_correlation.py
Comment (grlee77) [inline skimage/registration/tests/test_masked_phase_cross_correlation.py:27]:
```suggestion
```

This test case is no longer present in the `main` branch (This deprecated function was recently removed), but got reintroduced here.

**Review by grlee77 — CHANGES_REQUESTED** [review]
See suggestions on test case removal. This was the cause of the current CI failure.

The [removal was done in a recently merged PR](https://github.com/scikit-image/scikit-image/pull/5463/files#diff-88bfc1ae42e57a211c0cb8573895eed0521165fe1222e174e05bae070c2072cd)

**Comment by grlee77** [comment]
> Sorry for then noise. grlee already asked for a specific style. Lets follow it. This looks good to me.

No worries, thanks for reviewing!  For reference, we discussed this style choice previously during the SciPy sprint: https://github.com/scikit-image/scikit-image/pull/5478#pullrequestreview-708813111

**Review by grlee77 — APPROVED** [review]

**Comment by grlee77** [comment]
Thanks @Larkinnjm1, just waiting for CI to finish before merging

**Comment by grlee77** [comment]
The one CI failure that was observed is due to an unrelated PR that was merged and is resolved by #5522

**Pull request merged**
